gameoflife.py

```diff
@@ -76,25 +102,20 @@ def print_board(board):
         print('|')
 
 
-def main(args):
-    if len(args) != 2:
-        print("Please enter 2 arguments\n\t-file path to initial board state\n\t-number of generations")
-        exit()
-    try:
-        board = init_board(args[0])
-        generations = int(args[1])
-    except ValueError:
-        print(f"2nd argument must be a number, '{args[1]}' was entered.")
-        exit()
-    except Exception as e:
-        print(e)
-        exit()
-
+def main(file_path : str, generations : int) ->None:
+    board = init_board(file_path)
+    
     for i in range(generations):
         print_board(board)
         time.sleep(100/1000) # sleep for 100 ms
         board = advance_generation(board)
 
 
 if __name__ =="__main__":
-    main(sys.argv[1:])
\ No newline at end of file
+    parser = argparse.ArgumentParser(prog="Game Of Life",
+                                     description=description,
+                                     epilog="Try blinker.txt as an initial file for example, for as many generations you'd like")
+    parser.add_argument("file_path", help="the file path to the initial board state.")
```

Comment:
specify both arguments to be required=True

Reply:
the 'required' argument in add_argument is invalid for positional arguments and raises an error, but they are required by default.

Reply:
I know, read the docs and see how to define these as required and not optional (hint: the required flag is not enough)

https://docs.python.org/3/library/argparse.html